app/views/Project/Tagging/Export/NewExport/EntriesExportSelection/index.tsx

```diff
@@ -0,0 +1,481 @@
+import LeadsSelection from '../../LeadsSelection';
+import ExportTypePane from './ExportTypePane';
+
+import styles from './styles.css';
+
+const PROJECT_FRAMEWORK_DETAILS = gql`
+    query ProjectFrameworkDetails($projectId: ID!) {
+        project(id: $projectId) {
+            analysisFramework {
+                id
+                exportables {
+                    data
+                    id
+                    inline
+                    order
+                    widgetKey
+                    widgetType
+                    widgetTypeDisplay
+                }
+                primaryTagging {
+                    widgets {
+                        id
+                        clientId
+                        key
+                        order
+                        properties
+                        title
+                        widgetId
+                        width
+                    }
+                    clientId
+                    id
+                    order
+                    title
+                    tooltip
+                }
+                secondaryTagging {
+                    clientId
+                    id
+                    key
+                    order
+                    title
+                    properties
+                    widgetId
+                    width
+                }
+            }
+        }
+    }
+`;
+
+const CREATE_EXPORT = gql`
+    mutation CreateExport(
+        $projectId: ID!,
+        $data: ExportCreateInputType!,
+    ) {
+        project(id: $projectId) {
+            exportCreate(data: $data) {
+                ok
+                errors
+                result {
+                    id
+                    title
+                    isPreview
+                }
+            }
+        }
+    }
+`;
+
+const mapExportType: Record<ExportFormatEnum, ExportExportTypeEnum> = {
+    DOCX: 'REPORT',
+    PDF: 'REPORT',
+    JSON: 'JSON',
+    XLSX: 'EXCEL',
+};
+
+interface ExportReportStructure {
+    id: string;
+    levels?: ExportReportStructure[];
+}
+
+interface ReportStructureLevel {
+    id: string;
+    title: string;
+    sublevels?: ReportStructureLevel[];
+}
+
+const createReportStructureForExport = (nodes: Node[]): ExportReportStructure[] => (
+    nodes.filter((node) => node.selected)
+        .map((node) => ({
+            id: node.key,
+            ...(node.nodes && { levels: createReportStructureForExport(node.nodes) }),
+        }))
+);
+
+const createReportLevels = (nodes: Node[]): ReportStructureLevel[] => (
+    nodes
+        .filter((node) => node.selected)
+        .map((node) => ({
+            id: node.key,
+            title: node.title,
+            ...(node.nodes && { sublevels: createReportLevels(node.nodes) }),
+        }))
+);
+
+const createWidgetIds = (widgets: TreeSelectableWidget<string>[]) => (
+    widgets
+        .filter((widget) => widget.selected)
+        .map((widget) => +(widget.id))
```

Comment:
Any reason we have to cast this?
Do we not use graphql for all requests in export?